### PR TITLE
✨ Automate GitHub release creation

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Implementation of first phase of automated GitHub releases.

Test release on my CAPV fork: https://github.com/sbueringer/cluster-api-provider-vsphere/releases/tag/v1.8.0
Also verified all attached files and they look perfect.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

Part of #2051

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -9,17 +9,18 @@ require (
 	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230711141913-5dedc1b4f9d5
 	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20211110210527-619e6b92dab9
 	sigs.k8s.io/kind v0.17.0
-	sigs.k8s.io/kubebuilder-release-tools/notes v0.0.0-20220428224951-d8a44c7aef35
 	sigs.k8s.io/testing_frameworks v0.1.2
 )
 
+replace sigs.k8s.io/cluster-api/hack/tools => github.com/sbueringer/cluster-api/hack/tools v0.0.0-20230720105121-40ae6bbf3343
```
Comment (sbueringer) [inline hack/tools/go.mod:15]: Will be dropped once https://github.com/kubernetes-sigs/cluster-api/pull/9018 is merged and cherry-picked

Comment (randomvariable) [reply hack/tools/go.mod:15]: You'll do this before taking off WIP, right? If not, leave a comment.

**Comment by sbueringer** [comment]
Test release on my CAPV fork: https://github.com/sbueringer/cluster-api-provider-vsphere/releases/tag/v1.8.0
Also verified all attached files and they look perfect.

**Comment by sbueringer** [comment]
Only remainting task is now waiting for https://github.com/kubernetes-sigs/cluster-api/pull/9018 to be merged and cherry-picked

File: hack/tools/go.mod [hunk 2]
```diff
@@ -6,20 +6,19 @@ require (
 	github.com/joelanford/go-apidiff v0.6.0
 	github.com/onsi/ginkgo/v2 v2.11.0
 	k8s.io/code-generator v0.27.3
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230711141913-5dedc1b4f9d5
+	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230720182410-9c0a017bd4e3
```
Comment (sbueringer) [inline hack/tools/go.mod:9]: @chrischdi Let's make sure we bump this to the v1.5.0 commit next week. But I think for now it's fine to change this from the rc.0 commit to the latest commit of the release-1.5 release branch

**Comment by sbueringer** [comment]
/assign @randomvariable @srm09 @yastij 
Should be ready for review.

I will cherry-pick this back to all release branches manually. The versions of the relase note tool we use will slighty differ but that should be okay.

File: hack/tools/go.mod (hunk 1 above)
Comment (sbueringer) [reply hack/tools/go.mod]: Dropped the replace

**Pull request merged**